hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/FinalizeJoinOp.java

```diff
@@ -144,22 +155,28 @@ protected void writeInternalImpl(ObjectDataOutput out) throws IOException {
         out.writeLong(clusterStartTime);
         out.writeUTF(clusterState.toString());
         out.writeObject(clusterVersion);
+        if (clusterVersion.isGreaterOrEqual(V3_9) && preJoinOp != null) {
```

Comment:
why is there the `preJoinOp != null` in the `readInternalImpl()`  ? you always read the object during deserialization.